File: app/comment.rb
post '/comment/:comment_id/toggle_like' do |comment_id|
  require_login
  content_type :json
  comment = Comment[id: comment_id]
  liked_response = comment.toggle_site_like(current_site) ? 'liked' : 'unliked'
  {result: liked_response, comment_like_count: comment.comment_likes_dataset.count, liking_site_names: comment.liking_site_usernames}.to_json
end

post '/comment/:comment_id/delete' do |comment_id|
  require_login
  content_type :json
  comment = Comment[id: comment_id]

  if comment.event.site == current_site || comment.actioning_site == current_site
    comment.delete
    return {result: 'success'}.to_json
  end

  return {result: 'error'}.to_json
end

File: public/js/news/comment.js
var Comment = {
  create: function(eventId, csrfToken, form) {
    var form = $(form)
    var comment = form.find('[name="comment"]').val()
    form.remove()
    $.post('/event/'+eventId+'/comment', {csrf_token: csrfToken, message: comment}, function(res) {
      location.reload()
    })
  },

  delete: function(commentId, csrfToken) {
    var link = $('#comment_'+commentId+'_delete')

    if (link.hasClass('confirm')) {
      // user confirms deletion, delete
      $.post('/comment/'+commentId+'/delete', {csrf_token: csrfToken}, function(res) {
          location.reload()
      })
    } else {
      // first time user pressed delete, ask if they're sure
      var parentEl = link.parent()

      var spanEl = $('<span>Sure? </span>')

      link.addClass('confirm').text('Delete').detach()
      spanEl.append(link)

      spanEl.append($('<a>Cancel</a>').on('click', function(evnt) {
        var pEl = $(evnt.target.parentElement)
        var el = pEl.children('a.confirm').removeClass('confirm').text('Delete').detach()
        var ppEl = pEl.parent()
        pEl.remove()
        ppEl.append(el)
      }))

      parentEl.append(spanEl)
    }
  },

  toggleLike: function(commentId, csrfToken) {
    var link = $('#comment_'+commentId+'_like')

    $.post('/comment/'+commentId+'/toggle_like', {csrf_token: csrfToken}, function(res) {
      if(res.result == 'liked')
        link.text('Unlike ('+res.comment_like_count+')')

      if(res.result == 'unliked') {
        var linkText = 'Like'

        if(res.comment_like_count > 0)
          linkText += ' ('+res.comment_like_count+')'

        link.text(linkText)
      }
      link.attr('data-original-title', res.liking_site_names.join('<br>'))
    })
  }
}
